fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/QuerySegmentAggregator.java

```diff
@@ -555,20 +555,35 @@ protected void buildWhereClause(StringBuilder whereClause, String overrideType)
                     whereClause.append(whereClauseSegment);
                 } else {
                     if (!Type.COMPOSITE.equals(param.getType())) {
-                        // Join a standard parameter table
-                        //   JOIN Observation_TOKEN_VALUES AS param0
-                        //     ON param0.PARAMETER_NAME_ID=1191 AND param0.TOKEN_VALUE = :p1
-                        //    AND param0.LOGICAL_RESOURCE_ID = LR.LOGICAL_RESOURCE_ID
-
                         final String paramTableAlias = "param" + i;
-                        final String onFilter = querySegment.getQueryString().replaceAll(PARAMETER_TABLE_ALIAS + "\\.", paramTableAlias + ".");
-
-                        whereClause.append(JOIN);
-                        whereClause.append(tableName(overrideType, param));
-                        whereClause.append(" AS " + paramTableAlias);
-                        whereClause.append(ON);
-                        whereClause.append(onFilter);
-                        whereClause.append(" AND LR.LOGICAL_RESOURCE_ID = " + paramTableAlias + ".LOGICAL_RESOURCE_ID");
+                        if (param.isReverseChained()) {
+                            // Join on a select from logical resource table
+                            //   JOIN (
+                            //     SELECT CLR0.LOGICAL_ID FROM Observation_LOGICAL_RESOURCES AS CLR0
+                            //       ...
+                            //   ) AS param0 ON LR.LOGICAL_ID = param0.LOGICAL_ID
+                            whereClause.append(JOIN)
+                                        .append(LEFT_PAREN);
+                            whereClause.append(querySegment.getQueryString());
+                            whereClause.append(RIGHT_PAREN)
+                                        .append(" AS " + paramTableAlias)
```

Comment:
Just as a form, since we are using a StringBuilder let's build the string with it. 

```suggestion
                                        .append(" AS ").append(paramTableAlias)
```

Reply:
done